Dockerfile

```diff
@@ -48,4 +48,7 @@ COPY --from=build /app /app
 # This argument controls the value passed to npm start --workspace parameter
 ENV SERVICE=""
 
+# FIXME: don't invoke the service via shell
+# See https://github.com/filecoin-station/spark-api/issues/404
+SHELL ["/bin/bash", "-c"]
```

Comment:
Does this not still use a shell (`/bin/bash`)?

Reply:
That's a great question! I don't know what shell is executed by default. Maybe `/bin/sh` and perhaps that the one to use instead of Bash?

I was just following the advice given here:
https://docs.docker.com/reference/build-checks/json-args-recommended/

<img width="811" alt="Screenshot 2024-09-02 at 15 37 09" src="https://github.com/user-attachments/assets/b09df5da-cec9-45ff-9b67-d2549daa0747">

Reply:
I think this advice says that since you're already using a shell, it's better to be explicit about it

Reply:
Instead, we should fix the entry point

Reply:
I think it could be `CMD ['npm', 'start', '--workspace ${SERVICE}']`

Reply:
That won't work; Docker will not resolve the `SERVICE` provided via env var when building the arguments for CMD.

Reply:
I'm confused, isn't above `${SERVICE}=""`, so we could hardcode that value?

Reply:
I decided to bit the bullet and rework `SERVICE` to `NPM_CONFIG_WORKSPACE`

It seems to work, here is how I verified this change:

```
❯ docker run --rm -it --env=NPM_CONFIG_WORKSPACE=api spark-api

> @filecoin-station/spark-api@0.0.0 start
> node bin/spark.js

Warning: INFLUXDB_TOKEN was not provided by the environment
node:internal/process/esm_loader:34
      internalBinding('errors').triggerUncaughtException(
                                ^

AssertionError [ERR_ASSERTION]: GLIF_TOKEN must be provided in the environment variables
    at file:///app/common/ie-contract-config.js:8:1
    at ModuleJob.run (node:internal/modules/esm/module_job:222:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:323:24)
    at async loadESM (node:internal/process/esm_loader:28:7)
    at async handleMainPromise (node:internal/modules/run_main:113:12) {
  generatedMessage: false,
  code: 'ERR_ASSERTION',
  actual: false,
  expected: true,
  operator: '=='
}

Node.js v20.12.2
npm ERR! Lifecycle script `start` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: @filecoin-station/spark-api@0.0.0 
npm ERR!   at location: /app/api 
```